From 3de1687d7432ea9b302c2db9521996f506c140a3 Mon Sep 17 00:00:00 2001
From: Thomas De Schampheleire <thomas.de.schampheleire@gmail.com>
Date: Wed, 4 Nov 2015 11:07:32 +0100
Subject: [PATCH 102/120] mksquashfs: fix compilation on older toolchains
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Compiling mksquashfs on RHEL 5 (gcc 4.1.2, glibc 2.5) gives the following
error:

    mksquashfs.c: In function ‘get_fragment’:
    mksquashfs.c:1440: error: label at end of compound statement

On RHEL 6 (gcc 4.4.7, glibc 2.12), the preprocessed output shows a difference
for pthread_cleanup_pop, which now has a dummy do {} while (0); statement,
presumably to fix this exact error.

Fix for RHEL 5 by manually inserting that dummy statement.

Signed-off-by: Thomas De Schampheleire <thomas.de.schampheleire@gmail.com>
Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/mksquashfs.c        | 1 +
 squashfs-tools/process_fragments.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index d221c35..13d60f7 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -1411,6 +1411,7 @@ again:
 	cache_block_put(compressed_buffer);
 
 finished:
+	do { } while (0);
 	pthread_cleanup_pop(0);
 
 	return buffer;
diff --git a/squashfs-tools/process_fragments.c b/squashfs-tools/process_fragments.c
index bba6f5a..971dc11 100644
--- a/squashfs-tools/process_fragments.c
+++ b/squashfs-tools/process_fragments.c
@@ -210,6 +210,7 @@ again:
 	cache_block_put(compressed_buffer);
 
 finished:
+	do { } while (0);
 	pthread_cleanup_pop(0);
 
 	return buffer;
-- 
2.8.3

